executor/common_kvm_arm64_syzos.h

```diff
@@ -79,3 +92,37 @@ GUEST_CODE static void guest_uexit(uint64 exit_code)
 	volatile uint64* ptr = (volatile uint64*)ARM64_ADDR_UEXIT;
 	*ptr = exit_code;
 }
+
+#define GET_OP_FROM_REG(reg, field) \
+	((reg & KVM_REG_ARM64_SYSREG_##field##_MASK) >> KVM_REG_ARM64_SYSREG_##field##_SHIFT)
+
+// Generate an `MSR register, x0` instruction based on the register ID.
+GUEST_CODE static uint32 reg_to_msr(uint64 reg)
+{
+	uint32 opcode = 0xd5100000;
+	// Op0 is 2 bits, but the upper bit is always set to 1 by the opcode.
+	uint32 Op0 = GET_OP_FROM_REG(reg, OP0);
+	uint32 Op1 = GET_OP_FROM_REG(reg, OP1);
+	uint32 CRn = GET_OP_FROM_REG(reg, CRN);
+	uint32 CRm = GET_OP_FROM_REG(reg, CRM);
+	uint32 Op2 = GET_OP_FROM_REG(reg, OP2);
+	return opcode | (Op0 << 19) | (Op1 << 16) | (CRn << 12) | (CRm << 8) | (Op2 << 5);
+}
+
+// Write value to a system register using an MSR instruction.
+// The word "MSR" here has nothing to do with the x86 MSR registers.
+//
+// Prevent inlining this function, because we'll be doing a RET.
```

Comment:
This is not guaranteed to work. Maybe it works for you with -O2, but reproducers can be compiled with -O0 as well.
If the function creates a stack frame/adjusts SP, then JMP+RET won't restore that. + return address is not necessary in LR.

Reply:
Hm, on the other hand we can simply do BLR/RET

Reply:
Try to add bl to clobber list.